### PR TITLE
boost-python3: migrate to python@3.10

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

- boost-python3: migrate to python@3.10
- eigenpy: migrate to python@3.10
- graph-tool: migrate to python@3.10
- libtorrent-rasterbar: migrate to python@3.10
- networkit: migrate to python@3.10
- opencolorio: migrate to python@3.10
- openimageio: migrate to python@3.10
- pythran: migrate to python@3.10
- rdkit: migrate to python@3.10
- scipy: migrate to python@3.10
- versioned_dependencies_conflicts_allowlist: add `graph-tool`

### CODE REVIEW

File: Formula/graph-tool.rb [hunk 1]
```diff
@@ -84,13 +84,20 @@ class GraphTool < Formula
...
-    system "autoreconf", "-fiv"
-    xy = Language::Python.major_minor_version Formula["python@3.9"].opt_bin/"python3"
-    venv = virtualenv_create(libexec, Formula["python@3.9"].opt_bin/"python3")
+    system "autoreconf", "--force", "--install", "--verbose"
+    site_packages = Language::Python.site_packages(python3)
+    xy = Language::Python.major_minor_version(python3)
+    venv = virtualenv_create(libexec, python3)
 
     resources.each do |r|
       venv.pip_install_and_link r
```
Comment (carlocab) [inline Formula/graph-tool.rb:103]: This doesn't look right.

File: Formula/networkit.rb [hunk 2]
```diff
@@ -22,30 +23,25 @@ class Networkit < Formula
...
-    pth_contents = "import site; site.addsitedir('#{libexec/site_packages}')\n"
-    (prefix/site_packages/"homebrew-networkit.pth").write pth_contents
+    ENV.prepend_create_path "PYTHONPATH", prefix/site_packages
+    ENV.append_path "PYTHONPATH", Formula["libcython"].opt_libexec/site_packages
+
+    system "python3", "setup.py", "build_ext", "--networkit-external-core",
+                                               "--external-tlx=#{Formula["tlx"].opt_prefix}",
+                                               "--rpath=@loader_path;#{rpath_addons}"
+
+    system "python3", *Language::Python.setup_install_args
```
Comment (carlocab) [inline Formula/networkit.rb:40]:
```suggestion
    system "python3", *Language::Python.setup_install_args(prefix)
```

File: Formula/opencolorio.rb [hunk 3]
```diff
@@ -18,21 +19,18 @@ class Opencolorio < Formula
...
     ]
 
-    mkdir "macbuild" do
-      system "cmake", *args, ".."
-      system "make"
-      system "make", "install"
-    end
+    system "cmake", "-S", ".", "-B", "macbuild", *args, *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
```
Comment (cho-m) [inline Formula/opencolorio.rb:33]: Should probably be same directory rather than `macbuild` vs. `build`

Comment (carlocab) [reply Formula/opencolorio.rb:33]: Oops, yes, thanks

**Comment by carlocab** [comment]
I'm a bit suspicious about the `numpy` and `scipy` dependencies in `xgboost`. It tested fine, despite `python@3.10` not being in `PATH`.

**Comment by carlocab** [comment]
Yup, those dependencies are totally useless. #106980

File: Formula/graph-tool.rb (hunk 1 above)
Comment (cho-m) [reply Formula/graph-tool.rb:103]: Probably can clean up. Right now I don't think any `graph-tool` PyPI resource installs into `libexec/bin` so this isn't causing unwanted symlinks (i.e. `_and_link` is no-op).

**Comment by carlocab** [comment]
Looks like `openimageio` is already linking to `jpeg-turbo`:
```
==> brew linkage --cached openimageio
System libraries:
  /usr/lib/libSystem.B.dylib
  /usr/lib/libbz2.1.0.dylib
  /usr/lib/libc++.1.dylib
  /usr/lib/libz.1.dylib
Homebrew libraries:
  /opt/homebrew/opt/boost/lib/libboost_atomic-mt.dylib (boost)
  /opt/homebrew/opt/boost/lib/libboost_chrono-mt.dylib (boost)
  /opt/homebrew/opt/boost/lib/libboost_filesystem-mt.dylib (boost)
  /opt/homebrew/opt/boost/lib/libboost_system-mt.dylib (boost)
  /opt/homebrew/opt/boost/lib/libboost_thread-mt.dylib (boost)
  /opt/homebrew/opt/ffmpeg@4/lib/libavcodec.58.dylib (ffmpeg@4)
  /opt/homebrew/opt/ffmpeg@4/lib/libavformat.58.dylib (ffmpeg@4)
  /opt/homebrew/opt/ffmpeg@4/lib/libavutil.56.dylib (ffmpeg@4)
  /opt/homebrew/opt/ffmpeg@4/lib/libswscale.5.dylib (ffmpeg@4)
  /opt/homebrew/opt/freetype/lib/libfreetype.6.dylib (freetype)
  /opt/homebrew/opt/giflib/lib/libgif.dylib (giflib)
  /opt/homebrew/opt/imath/lib/libImath-3_1.29.dylib (imath)
  /opt/homebrew/opt/jpeg-turbo/lib/libjpeg.8.dylib (jpeg-turbo)
  /opt/homebrew/opt/libheif/lib/libheif.1.dylib (libheif)
  /opt/homebrew/opt/libpng/lib/libpng16.16.dylib (libpng)
  /opt/homebrew/opt/libraw/lib/libraw_r.20.dylib (libraw)
  /opt/homebrew/opt/libtiff/lib/libtiff.5.dylib (libtiff)
  /opt/homebrew/opt/opencolorio/lib/libOpenColorIO.2.1.dylib (opencolorio)
  /opt/homebrew/opt/openexr/lib/libIex-3_1.30.dylib (openexr)
  /opt/homebrew/opt/openexr/lib/libIlmThread-3_1.30.dylib (openexr)
  /opt/homebrew/opt/openexr/lib/libOpenEXR-3_1.30.dylib (openexr)
  /opt/homebrew/opt/openexr/lib/libOpenEXRCore-3_1.30.dylib (openexr)
  /opt/homebrew/opt/webp/lib/libwebp.7.dylib (webp)
  /opt/homebrew/opt/webp/lib/libwebpdemux.2.dylib (webp)
Indirect dependencies with linkage:
  jpeg-turbo
Variable-referenced libraries:
  @rpath/libOpenImageIO.2.3.dylib
  @rpath/libOpenImageIO_Util.2.3.dylib
Dependencies with no linkage:
  boost-python3
```
May as well migrate it. Hopefully that doesn't break anything.

**Comment by carlocab** [comment]
Actually, maybe I'll hold off on that. Probably good to avoid the problems seen at https://github.com/Homebrew/homebrew-core/pull/107244.

**Comment by cho-m** [comment]
I think it should be fine as is for now and Jpeg migration can be handled in other PR. It looks like `openimageio` links to versioned library, which should avoid any odd mixing from things like Linux RPATH ordering.

**Comment by carlocab** [comment]
> It looks like `openimageio` links to versioned library, which should avoid any odd mixing from things like Linux RPATH ordering.

This will always happen on macOS because of the library's install name:
```
❯ otool -D "$(brew --prefix jpeg-turbo)/lib/libjpeg.dylib"
/usr/local/opt/jpeg-turbo/lib/libjpeg.dylib:
/usr/local/opt/jpeg-turbo/lib/libjpeg.8.dylib
```
Not sure how it works on Linux.

**Review by cho-m — APPROVED** [review]

**Pull request closed without merging**